### DISCUSSION TITLE
Force Combination for subfolders?

### DISCUSSION THREAD

**Discussion by echo-of-machines** [nftchef/art-engine#45] [Help + Q&A]
Is it possible to force combinations on folders?
For example, if I have a "Fire Hair" folder with different layers, and I want this specific folder to combine with only "Fire Body", which also contains a bunch of layers, is this doable?

**Comment by nftchef** [comment]
Hi,
ForcedCombinations is only intended for single file combinations. This can and should be done by organizing subfolders into branches.
<img width="260" alt="image" src="https://user-images.githubusercontent.com/91582112/155136638-c7db6343-f207-4ffe-9852-e32938aed426.png">

hope that helps!

**Reply by echo-of-machines** [reply]
Thanks for the answer, that helps a ton :)
But I ran into another problem.
Using the example above, if I need OptionA from the folder "body" to OptionA in "Hair" when using the force combination, there will be times where OptionA from "Hair" would appear alone without OptionA from "body" in some of the images because of folder order, and combining doesn't work backward.  Is there a solution to that?

**Reply by echo-of-machines** [reply]
For example, if I type
OptionA:["OptionA"], this would let the first OptionA from the folder "Body" to force combine with OptionA in "Hair" thus successfully combining the two.  But if the combination didn't pick OptionA from "Body", there will be times where I'll have images that only have OptionA from "Hair".  In the situation where OptionA must only appear with OptionA, is there a solution?

**Reply by nftchef** [reply]
if the combination is truly a 1:1 forced combination, then you can use forcedCombinations, but yes, you would need to adjust the `layersOrder` to make sure they are chosen in the correct order. 

if you have more options, then you can branch the folders "one more branch" deep, using folders as "if,then" decisions
something like
<img width="303" alt="image" src="https://user-images.githubusercontent.com/91582112/155178587-5c01a7cd-aecf-4136-b5ea-6c17580a7f89.png">

**Reply by soulmv** [reply]
There is also a workaround - setting incompatible layers. I know it's too much code but it worked for me.
For example you ve got  two folders with Option A B C in one and Option D E F in another, and you want to pick Options A and E together. It should look like this in incompatible layers code:

OptionA: [ " OptionD", " OptionF" ]

That sets incompatibility between A and D F options, kind of reversal method.
But if you have 20 items in each folder - that might become a bit of pain to set up :)

Hope that helps.

**Reply by echo-of-machines** [reply]
> 

This would still cause the same problem where the folder with D E F comes after the folder with A B C. This means E can't check if it's paired with A.  So once we get to the D E F folder, we'll have E appear without A.  In other words, OptionE: ["OptionB, "OptionC"] won't work because of the layer structure.
